### ISSUE TITLE
Download release assets 

### ISSUE THREAD

**Issue by nicola-lunghi** [PyGithub/PyGithub#2918]
The documentation says that I can download a release assets with:
- https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#get-a-release-asset

```
To download the asset's binary content, set the Accept header of the request to [application/octet-stream](https://docs.github.com/rest/overview/media-types). The API will either redirect the client to the location, or stream it directly if possible. API clients should handle both a 200 or 302 response.
```

Can I do it with this library?

**Comment by xvega** [comment]
> The documentation says that I can download a release assets with:
> 
> * https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#get-a-release-asset
> 
> ```
> To download the asset's binary content, set the Accept header of the request to [application/octet-stream](https://docs.github.com/rest/overview/media-types). The API will either redirect the client to the location, or stream it directly if possible. API clients should handle both a 200 or 302 response.
> ```
> 
> Can I do it with this library?

I don't think this library provides a direct method to download assets,  but you could use the asset's URL with a HTTP library like `requests` to download the asset.

https://github.com/PyGithub/PyGithub/blob/0b8435fccbcc98404f08e146b6e259bd20065c98/github/GitReleaseAsset.py#L126

This is an idea/pseudocode that might help you get started (haven't tested this)

```python
from github import Github
import requests

g = Github("your token")
repo = g.get_repo("your repo")

# Get the specific release or terate over releases
release = repo.get_release("tag_name")

# Loop through assets in the release
for asset in release.get_assets():
    response = requests.get(asset.browser_download_url, allow_redirects=True)
    # whatever you need to do with the binary 
```

**Comment by nicola-lunghi** [comment]
i solved with something similar but was wondering if could be added

**Comment by nicola-lunghi** [comment]
Hi @EnricoMi if I submit a PR would you accept this? there's a good reason why this is not included? there's a similar API also for downloading job artifacts

**Comment by neel-m** [comment]
There is some history:
https://github.com/PyGithub/PyGithub/issues/1541
https://github.com/PyGithub/PyGithub/issues/2445

There was a pull request:
https://github.com/PyGithub/PyGithub/pull/2624

While the browser_download_url might work for a public repository, it definitely won't for a private repo which needs proper authentication headers.

**Comment by EnricoMi** [comment]
@nicola-lunghi yes, please, go ahead!

**Comment by neel-m** [comment]
I had a local change for `download_asset` and `requestFile` that worked on code from a year or so ago.  I can clean it up and submit a PR, if someone will look at it.

**Comment by EnricoMi** [comment]
Will do.

**Comment by neel-m** [comment]
Looks like there have been some changes to the repo, so will need to refactor my change to work with the changed code.

**Comment by neel-m** [comment]
First draft is ready here: https://github.com/PyGithub/PyGithub/pull/3060

**Comment by neel-m** [comment]
Any comments on the PR?
What is the process for review?
It looks like the project is anti- https://www.conventionalcommits.org/en/v1.0.0/ if so, it would be good to put that in the contribution guide.
